### PR TITLE
Fix marking 'fi;' as executed in profiles

### PR DESCRIPTION
**Pull request by ChrisJefferson**
This fixes 'fi;' in profiles. There are no tests, I will add tests to the 'profiling' package (but by inspection, it seems to produce the right answer).

### CODE REVIEW

**Comment by fingolfin** [comment]
Awesome. Now I wonder would the `od` in `for` and `while` loops possibly be affected, too?

**Comment by ChrisJefferson** [comment]
They seem to be fine, because in `while` and `repeat` the interiors are (a) coded, and (b) never ignored. There might be a similar bug if someone wrote a multi-line `Info` or `Assert` statement, where we also try to ignore bits of code, but long multi-line statements sometimes slip up profiling already.

File: src/intrprtr.c [hunk 1]
```diff
@@ -3921,8 +3928,12 @@ void            IntrInfoEnd( UInt narg )
...
 
     /* ignore or code                                                      */
-    SKIP_IF_RETURNING();
-    if ( STATE(IntrIgnoring)  > 1 ) { STATE(IntrIgnoring)--; return; }
+     INTERPRETER_PROFILE_HOOK(1);
+     SKIP_IF_RETURNING_NO_PROFILE_HOOK();
+
+     if (STATE(IntrIgnoring) > 1) {
+         STATE(IntrIgnoring)--;
+         return; }
```
Comment (fingolfin) [inline src/intrprtr.c]: I'd either keep this on one line, as it was; or else fully clang-format it (and get that brace on its own line)

File: src/intrprtr.c [hunk 2]
```diff
@@ -659,15 +663,18 @@ Int            IntrIfEndBody (
...
+    SKIP_IF_RETURNING_NO_PROFILE_HOOK();
+
     if ( STATE(IntrIgnoring)  > 1 ) { STATE(IntrIgnoring)--; return; }
 
     // if one branch was executed (ignoring the others), reset IntrIgnoring
     if ( STATE(IntrIgnoring) == 1 ) {
         STATE(IntrIgnoring) = 0;
     }
 
+
```
Comment (fingolfin) [inline src/intrprtr.c]: Remove extra empty line?

File: src/intrprtr.c [hunk 3]
```diff
@@ -4023,8 +4034,11 @@ void             IntrAssertAfterCondition ( void )
 void             IntrAssertEnd2Args ( void )
 {
       /* ignore or code                                                      */
-    SKIP_IF_RETURNING();
-    if ( STATE(IntrIgnoring)  > 2 ) { STATE(IntrIgnoring) -= 2; return; }
+      INTERPRETER_PROFILE_HOOK(2);
+      SKIP_IF_RETURNING_NO_PROFILE_HOOK();
+      if (STATE(IntrIgnoring) > 2) {
+          STATE(IntrIgnoring) -= 2;
+          return; }
```
Comment (fingolfin) [inline src/intrprtr.c]: see above

File: src/intrprtr.c [hunk 4]
```diff
@@ -4023,8 +4034,11 @@ void             IntrAssertAfterCondition ( void )
 void             IntrAssertEnd2Args ( void )
 {
       /* ignore or code                                                      */
-    SKIP_IF_RETURNING();
-    if ( STATE(IntrIgnoring)  > 2 ) { STATE(IntrIgnoring) -= 2; return; }
+      INTERPRETER_PROFILE_HOOK(2);
```
Comment (fingolfin) [inline src/intrprtr.c]: Huh, the indentation looks off, at least compared to what was there before?!?

File: src/intrprtr.c (hunk 1 above)
Comment (ChrisJefferson) [reply src/intrprtr.c]: Weirdly, this was clang-formatted. However, clang-format seems to have not reached a fixed point, and re-clang-formatting changes it again.

File: src/intrprtr.c (hunk 4 above)
Comment (ChrisJefferson) [reply src/intrprtr.c]: the old formatting was 2 space (which nothing else in this file is), that's clang-format moving it to 4 spaces.

**Comment by ChrisJefferson** [comment]
Now made all formatting (hopefully) reasonable, and locally consistent.

**Comment by alex-konovalov** [comment]
@ChrisJefferson I am puzzled that the label backport-4.10 has been removed, but this is still assigned to the milestone 4.10.1. I think that the initial intention was to backport it, but then it was decided to not to do it - in this case, both the label and the milestone should be updated.

**Comment by fingolfin** [comment]
This is not needed in the release notes, because it fixes a bug in a feature that was newly added (i.e., does not exist in 4.10)

**Pull request merged**